path_login.go

```diff
@@ -167,7 +178,15 @@ func (b *jwtAuthBackend) pathLogin(ctx context.Context, req *logical.Request, d
 			Time:    time.Now(),
 		}
 
-		if err := claims.Validate(expected); err != nil {
+		// We bake leeway into expiration/not before, so we pass in a duration of 0
+		// to account for our implementation.  Default to JWT's package of 1 minute
+		// if no leeways are configured
+		leeway := time.Second * 0
+		if role.ExpirationLeeway == 0 && role.NotBeforeLeeway == 0 {
+			leeway = jwt.DefaultLeeway
```

Comment:
This feels odd to me -- this means that if you don't set either of those, we're adding in 5 minutes of leeway above (in each direction), but then another minute here, so it's actually a 12 minute validity period. Granted this leeway accounts for clock skew in either direction, which the above calculations don't, but in the same vein, if you define the leeways above, or if you have `iat`/`nbf` defined, now you're *not* accounting for clock skew.

I wonder if the right logic is that we should always use the library's default leeway value to account for clock skew, but reduce the defaults we use above from 300 seconds down to, say, 150 seconds for each. That gives a validity window of 300 seconds total plus leeway.

If we add leeway all the time we should document it in CHANGES as it's a behavioral change. It may be worth another parameter, to be honest -- something like `clock_skew_leeway` that gets applied regardless of claims and in both directions, where we default to the library's default but people can turn it off if they don't want an extra minute of validity (which they may well not want).